libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java

```diff
@@ -154,6 +186,10 @@ public BytesRef readBytesRef(int length) throws IOException {
         return new BytesRef(bytes, 0, length);
     }
 
+    public void readFully(byte[] b) throws IOException {
```

Comment:
This method won't consume full stream, it will consume up to `b` length, I think `readBytes` (overloaded version) would be a better name: 
```
public void readBytes(byte[] b)
```